### PR TITLE
✨ Notifications: first draft

### LINKED ISSUE #3: implement push notifications
https://documentation.onesignal.com/docs/react-native-expo-sdk-setup

### PR DESCRIPTION
**Pull request by jgalat**
- Introduces One Signal for notifications
- Add in-app subscription (external id still needs to be setup)
- Backend utilities for sending
- Adds Pomelo notifications endpoint
- Some draft icons to use in notifications
- OneSignal SDK service worker for web notifications. ~DISCLAIMER: I still couldn't make web notifications work. For some reason when running `bun web` or exporting (`bunx expo export --platform web`) and serving, the html in `+html.tsx` is being ditched and a default template is being used. Been trying to fix that..~
- Metro: Ignore `/contracts/lib/` path

closes #3

### CODE REVIEW

File: app.config.ts [hunk 1]
```diff
@@ -28,6 +29,14 @@ export default {
     ],
     "expo-router",
     "sentry-expo",
+    [
+      "onesignal-expo-plugin",
+      {
+        mode: "development",
+        smallIcons: ["./assets/ic_stat_onesignal_default.png"],
+        largeIcons: ["./assets/ic_onesignal_large_icon_default.png"],
+      } as OneSignalPluginProps,
```
Comment (sebipap) [inline app.config.ts:38]: maybe could use `satisfies`

File: package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
     "@alchemy/aa-core": "^1.2.0",
     "@exactly/protocol": "^0.2.19",
     "@expo/vector-icons": "^13.0.0",
+    "@onesignal/node-onesignal": "^2.0.1-beta2",
```
[the remark this answers is not in the document]
Comment (cruzdanilo) [reply package.json:32]: literally, all versions are beta 😂
https://www.npmjs.com/package/@onesignal/node-onesignal?activeTab=versions

File: pomelo/api/transactions/v1/notifications.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        headings: {
+          en: event.status,
+        },
+        contents: {
+          en: event.status_detail,
+        },
+      });
+      return signResponse(request, response.status(200), JSON.stringify(true));
+    } catch {
+      // TODO(jg): log errors..
```
[the remark this answers is not in the document]
Comment (jgalat) [reply pomelo/api/transactions/v1/notifications.ts:44]: Definitely, we need to setup sentry for this backend

File: app.config.ts [hunk 4]
```diff
@@ -28,6 +29,14 @@ export default {
     ],
     "expo-router",
     "sentry-expo",
+    [
+      "onesignal-expo-plugin",
+      {
+        mode: "development",
```
[the remark this answers is not in the document]
Comment (jgalat) [reply app.config.ts:35]: For some reason i am getting an error when trying to import the enum from `"onesignal-expo-plugin/types/types"`

[the remark this answers is not in the document]
Comment (jgalat) [reply app.config.ts:35]: Ok. Found the reason. Given that i am importing `Mode` as a value, I cannot use the same import file.
I had to use the js compiled version `"onesignal-expo-plugin/build/types/types"`, which comes untyped (`any`), which then in order to be used with the `satisfies` keyword i have to cast it to its type. At the same time I have 2 eslint rules to ignore

image
image
image

**Review by sebipap — APPROVED** [review]

**Review by cruzdanilo — APPROVED** [review]

**Pull request merged**